inference-engine/ie_bridges/python/src/openvino/inference_engine/ie_api_impl.cpp

```diff
@@ -665,6 +668,163 @@ void InferenceEnginePython::IECore::addExtension(const std::string& ext_lib_path
+        impl = PyObject_CallObject(def, args);
+        Py_DECREF(data);
+        Py_DECREF(args);
+
+        PyGILState_Release(gstate);
+
+        inpShapes.resize(node->get_input_size());
+        for (int i = 0; i < inpShapes.size(); ++i) {
+            inpShapes[i] = node->get_input_shape(i);
+        }
+
+        outShapes.resize(node->get_output_size());
+        for (int i = 0; i < outShapes.size(); ++i) {
+            outShapes[i] = node->get_output_shape(i);
+        }
+    }
+
+    InferenceEngine::StatusCode getSupportedConfigurations(std::vector<InferenceEngine::LayerConfig>& conf,
+                                                           InferenceEngine::ResponseDesc* resp) noexcept override {
+        std::vector<InferenceEngine::DataConfig> inDataConfig;
+        std::vector<InferenceEngine::DataConfig> outDataConfig;
+
+        // Allow any offset before data
+        size_t offset((std::numeric_limits<size_t>::max)());
+
+        // Input shapes
+        for (auto& inpShape : inpShapes) {
+            InferenceEngine::SizeVector order(inpShape.size());
+            std::iota(order.begin(), order.end(), 0);
+
+            InferenceEngine::DataConfig inpConf;
+            inpConf.desc = InferenceEngine::TensorDesc(InferenceEngine::Precision::FP32, inpShape, {inpShape, order, offset});
+            inDataConfig.push_back(inpConf);
+        }
+
+        // Output shapes
+        for (auto& outShape : outShapes) {
+            InferenceEngine::SizeVector order(outShape.size());
+            std::iota(order.begin(), order.end(), 0);
+
+            InferenceEngine::DataConfig outConf;
+            outConf.desc = InferenceEngine::TensorDesc(InferenceEngine::Precision::FP32, outShape, {outShape, order, offset});
+            outDataConfig.push_back(outConf);
+
+        }
+        InferenceEngine::LayerConfig layerConfig;
+        layerConfig.inConfs = inDataConfig;
+        layerConfig.outConfs = outDataConfig;
+
+        conf.push_back(layerConfig);
+
+        return InferenceEngine::StatusCode::OK;
+    }
+
+    InferenceEngine::StatusCode init(InferenceEngine::LayerConfig& config,
+                                     InferenceEngine::ResponseDesc* resp) noexcept override {
+        return InferenceEngine::OK;
+    }
+
+    InferenceEngine::StatusCode execute(std::vector<InferenceEngine::Blob::Ptr>& inputs,
+                                        std::vector<InferenceEngine::Blob::Ptr>& outputs,
+                                        InferenceEngine::ResponseDesc* resp) noexcept override {
+        _import_array();
+
+        PyObject* args = PyList_New(inputs.size());
+        for (size_t i = 0; i < inputs.size(); ++i) {
+            void* data = inputs[i]->buffer();
+            std::vector<size_t> dims = inputs[i]->getTensorDesc().getDims();
+            std::vector<npy_intp> npy_dims(dims.begin(), dims.end());
+            PyObject* arr = PyArray_SimpleNewFromData(dims.size(), npy_dims.data(), NPY_FLOAT32, data);
```

Comment:
Do we have alternative C++ -> numpy conversion?

Reply:
Note: is it always float32 data type?

CC @akuporos